### DISCUSSION TITLE
Extending admin console

### DISCUSSION THREAD

**Discussion by Captain-P-Goldfish** [keycloak/keycloak#14327] [Archive]
I found the page navigation within the keycloak-ui project and I do not see any possibility to extend it anymore. Am I correct that this feature will no longer be supported?

```js
return (
    <PageSidebar
      className="keycloak__page_nav__nav"
      nav={
        <Nav onSelect={onSelect}>
          <NavList>
            <NavItem className="keycloak__page_nav__nav_item__realm-selector">
              <RealmSelector />
            </NavItem>
          </NavList>
          <Divider />
          {showManage && !isOnAddRealm && (
            <NavGroup aria-label={t("manage")} title={t("manage")}>
              <LeftNav title="clients" path="/clients" />
              <LeftNav title="clientScopes" path="/client-scopes" />
              <LeftNav title="realmRoles" path="/roles" />
              <LeftNav title="users" path="/users" />
              <LeftNav title="groups" path="/groups" />
              <LeftNav title="sessions" path="/sessions" />
              <LeftNav title="events" path="/events" />
            </NavGroup>
          )}

          {showConfigure && !isOnAddRealm && (
            <NavGroup aria-label={t("configure")} title={t("configure")}>
              <LeftNav title="realmSettings" path="/realm-settings" />
              <LeftNav title="authentication" path="/authentication" />
              <LeftNav title="identityProviders" path="/identity-providers" />
              <LeftNav title="userFederation" path="/user-federation" />
            </NavGroup>
          )}
        </Nav>
      }
    />
  );
```

**Comment by ssilvert** [comment]
Exactly what feature are you referring to?  What are you trying to accomplish?

We are in discussions about how to make it easier to extend and modify both the new admin console and a rewrite of the account console.  So feedback about your use case will be helpful.

**Comment by Captain-P-Goldfish** [comment]
this is actually referring to the scim-for-keycloak project "https://github.com/Captain-P-Goldfish/scim-for-keycloak". In the old admin console I added a new menu link to provide additional configuration pages for SCIM.

**Comment by ssilvert** [comment]
The solution for now is to fork the keycloak-ui repo and build your own complete version of admin console.  But this will be hard to maintain, so we are thinking about a better way to do it.  

Thanks for your feedback.  It's helpful to look at your use case.

**Reply by kkcmadhu** [reply]
any update on this , I was  looking at using the SCIM plugin, but i think this  will need to be fixed prior that :(

**Reply by BradNut** [reply]
@ssilvert Also wondering if there's any update on this. It would not be ideal for the plugin maintainer to build a complete version or the users of a plugin to do the same.

**Comment by ssilvert** [comment]
@kkcmadhu @BradNut All I can tell you is that we are still planning to address this but we've had bigger priorities.  I'm hoping that will change soon.  Sorry I can't give you a better answer at this time.

**Reply by jonkoops** [reply]
@ssilvert just FYI, you can reply inside of a thread. There is no need to start a new one.

**Reply by keerthivibisan** [reply]
any update on this, To extend and modify the new admin console.

**Reply by ssilvert** [reply]
Sorry it is taking so long to get this project moving.  We did just release a list of proposed requirements.  Feel free to add your feedback.
https://github.com/keycloak/keycloak/discussions/22255

**Comment by DaMo86** [comment]
Since merging the separate Admin-UI into the Mono-Repo, the suggested way of forking the separate Repo is no longer working.

Do you have already further ideas about easier integration/extending the admin console @ssilvert ?

**Reply by ssilvert** [reply]
Just fork the mono-repo.

**Reply by DaMo86** [reply]
Thank you @ssilvert 
Forking and maintaining the complete mono repo just for the adminUI sounds like a hazzle, we would be really interested in leaner options here.
